### ISSUE TITLE
V8 'getTransaction` return is different than 'streamTransactions' for the same TxId

### ISSUE THREAD

**Issue by osagga** [bitpay/bitcore#1749]
I'm seeing this weird error (***only recently after the recent updates***) where the transaction returned by `/api/BTC/regtest/tx/[txId]` is different (also incorrect) than the one returned by `/api/BTC/regtest/tx/?blockHeight=[blockHeight]` with the blockHeight that the BTC full node reports as the one that includes the `txId`. 

I'm not really sure what could be causing this, it seems that the Db for some reason has two copies of the `txId` object, one that shows that the Tx is not mined yet (in the mempool), and the other correct one that is linked to the block.

```json
$ curl -s localhost:3000/api/BTC/regtest/tx/?blockHeight=1253 | python -m json.tool
[
    {
        "_id": "5bd3921f5dfb111725b75218",
        "blockHash": "6389e3068b3d907c23c567c03ef145cc243ae9272be704807da3c9a892f784d0",
        "blockHeight": 1253,
        "blockTime": "2018-10-26T22:16:07.000Z",
        "blockTimeNormalized": "2018-10-26T22:16:07.002Z",
        "coinbase": true,
        "confirmations": 6,
        "fee": 0,
        "locktime": 0,
        "network": "regtest",
        "raw": "02000000010000000000000000000000000000000000000000000000000000000000000000ffffffff0502e5040101ffffffff028a1d2a01000000002321024607f6a541b3b7d091b41c4aa3456086d9636deb4bf079e63966a0b6dd342aa4ac0000000000000000266a24aa21a9ed259ffc76adb7588a6d416cc6dd69e80a34c1c4080e180f8cf17a7b90f4ddd57700000000",
        "size": 147,
        "txid": "2457c85aa6d6c13773cf061ac301a8ff730449a0ebb273b1dae1974d88e987f2"
    },
    {
        "_id": "5bd3921f5dfb111725b75219",
        "blockHash": "6389e3068b3d907c23c567c03ef145cc243ae9272be704807da3c9a892f784d0",
        "blockHeight": 1253,
        "blockTime": "2018-10-26T22:16:07.000Z",
        "blockTimeNormalized": "2018-10-26T22:16:07.002Z",
        "coinbase": false,
        "confirmations": 6,
        "fee": 6040,
        "locktime": 1252,
        "network": "regtest",
        "raw": "02000000025e3365fee509d7e21b466042c0ffc193e4d3e6e6b197825516c54c71af93e9730000000049483045022100ea42594ec708c826bec32d9e1f1acd1a1d733755665ab512a387871f3620ee8e022041ed5c08aa17138f631748b7ee94ecd18e35df4788c5b340f6445116da5e444201feffffffc0c7a77db259d047889cd8dd2289aaa5a2dadf836b17979df26b7607354b93490000000049483045022100a36be156d8184ce4931e0bb73b99b8b2de0f4baf7262bcb0a61462b63882b6b4022032954210cf805f091ea6b1c3827c2e6ceca03d7df7baa94ea8d2c32544d7ab0b01feffffff02d5f66a0a0000000017a9149207eb49b1984bb30f732026f5e698a1d2adc64d87072d39010000000017a91492f9765429e7aaae39d0d67c2b4565f3a699f4f287e4040000",
        "size": 302,
        "txid": "17dcb29f6ae6821b398a8bd9fe04452f99c8320f29ea19ab077363e7d2113d63"
    }
]

$ curl -s localhost:3000/api/BTC/regtest/tx/17dcb29f6ae6821b398a8bd9fe04452f99c8320f29ea19ab077363e7d2113d63 | python -m json.tool
[
    {
        "_id": "5bd3921f5dfb111725b7520a",
        "blockHash": null,
        "blockHeight": -1,
        "blockTime": "2018-10-26T22:15:59.268Z",
        "blockTimeNormalized": "2018-10-26T22:15:59.268Z",
        "coinbase": false,
        "confirmations": 0,
        "fee": 6040,
        "locktime": 1252,
        "network": "regtest",
        "raw": "02000000025e3365fee509d7e21b466042c0ffc193e4d3e6e6b197825516c54c71af93e9730000000049483045022100ea42594ec708c826bec32d9e1f1acd1a1d733755665ab512a387871f3620ee8e022041ed5c08aa17138f631748b7ee94ecd18e35df4788c5b340f6445116da5e444201feffffffc0c7a77db259d047889cd8dd2289aaa5a2dadf836b17979df26b7607354b93490000000049483045022100a36be156d8184ce4931e0bb73b99b8b2de0f4baf7262bcb0a61462b63882b6b4022032954210cf805f091ea6b1c3827c2e6ceca03d7df7baa94ea8d2c32544d7ab0b01feffffff02d5f66a0a0000000017a9149207eb49b1984bb30f732026f5e698a1d2adc64d87072d39010000000017a91492f9765429e7aaae39d0d67c2b4565f3a699f4f287e4040000",
        "size": 302,
        "txid": "17dcb29f6ae6821b398a8bd9fe04452f99c8320f29ea19ab077363e7d2113d63"
    }
]

$ ~/bitcore-v8/full-nodes/BTC/bitcoin-0.16.3/bin/bitcoin-cli -datadir=/home/corvus/bitcore-v8/full-nodes/BTC/data/ getrawtransaction 17dcb29f6ae6821b398a8bd9fe04452f99c8320f29ea19ab077363e7d2113d63 true
{
  "txid": "17dcb29f6ae6821b398a8bd9fe04452f99c8320f29ea19ab077363e7d2113d63",
  "hash": "17dcb29f6ae6821b398a8bd9fe04452f99c8320f29ea19ab077363e7d2113d63",
  "version": 2,
  "size": 302,
  "vsize": 302,
  "locktime": 1252,
  "vin": [
    {
      "txid": "73e993af714cc516558297b1e6e6d3e493c1ffc04260461be2d709e5fe65335e",
      "vout": 0,
      "scriptSig": {
        "asm": "3045022100ea42594ec708c826bec32d9e1f1acd1a1d733755665ab512a387871f3620ee8e022041ed5c08aa17138f631748b7ee94ecd18e35df4788c5b340f6445116da5e4442[ALL]",
        "hex": "483045022100ea42594ec708c826bec32d9e1f1acd1a1d733755665ab512a387871f3620ee8e022041ed5c08aa17138f631748b7ee94ecd18e35df4788c5b340f6445116da5e444201"
      },
      "sequence": 4294967294
    },
    {
      "txid": "49934b3507766bf29d97176b83dfdaa2a5aa8922ddd89c8847d059b27da7c7c0",
      "vout": 0,
      "scriptSig": {
        "asm": "3045022100a36be156d8184ce4931e0bb73b99b8b2de0f4baf7262bcb0a61462b63882b6b4022032954210cf805f091ea6b1c3827c2e6ceca03d7df7baa94ea8d2c32544d7ab0b[ALL]",
        "hex": "483045022100a36be156d8184ce4931e0bb73b99b8b2de0f4baf7262bcb0a61462b63882b6b4022032954210cf805f091ea6b1c3827c2e6ceca03d7df7baa94ea8d2c32544d7ab0b01"
      },
      "sequence": 4294967294
    }
  ],
  "vout": [
    {
      "value": 1.74782165,
      "n": 0,
      "scriptPubKey": {
        "asm": "OP_HASH160 9207eb49b1984bb30f732026f5e698a1d2adc64d OP_EQUAL",
        "hex": "a9149207eb49b1984bb30f732026f5e698a1d2adc64d87",
        "reqSigs": 1,
        "type": "scripthash",
        "addresses": [
          "2N6ZN8X5e5Y84gYuoMophbSiBb712ZXcb2Q"
        ]
      }
    },
    {
      "value": 0.20524295,
      "n": 1,
      "scriptPubKey": {
        "asm": "OP_HASH160 92f9765429e7aaae39d0d67c2b4565f3a699f4f2 OP_EQUAL",
        "hex": "a91492f9765429e7aaae39d0d67c2b4565f3a699f4f287",
        "reqSigs": 1,
        "type": "scripthash",
        "addresses": [
          "2N6eMVEJjdtpu2x8Mqgno5Lhi1NxhJuz79c"
        ]
      }
    }
  ],
  "hex": "02000000025e3365fee509d7e21b466042c0ffc193e4d3e6e6b197825516c54c71af93e9730000000049483045022100ea42594ec708c826bec32d9e1f1acd1a1d733755665ab512a387871f3620ee8e022041ed5c08aa17138f631748b7ee94ecd18e35df4788c5b340f6445116da5e444201feffffffc0c7a77db259d047889cd8dd2289aaa5a2dadf836b17979df26b7607354b93490000000049483045022100a36be156d8184ce4931e0bb73b99b8b2de0f4baf7262bcb0a61462b63882b6b4022032954210cf805f091ea6b1c3827c2e6ceca03d7df7baa94ea8d2c32544d7ab0b01feffffff02d5f66a0a0000000017a9149207eb49b1984bb30f732026f5e698a1d2adc64d87072d39010000000017a91492f9765429e7aaae39d0d67c2b4565f3a699f4f287e4040000",
  "blockhash": "6389e3068b3d907c23c567c03ef145cc243ae9272be704807da3c9a892f784d0",
  "confirmations": 6,
  "time": 1540592167,
  "blocktime": 1540592167
}



```

**Comment by micahriggan** [comment]
Hey Omar, thanks for catching that! We switched a few things to inserts
rather than updates with uosert, but it turns out that results in
duplication like you mentioned. We'll get that reverted.

**Comment by osagga** [comment]
It seems fixed now with the commit f3dad47022432d72215ffd983b02a11dd9bd05f8, thanks!
